shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -198,7 +210,7 @@ - (void)setTextInputState:(NSDictionary*)state {
   if (selectedRange.location != oldSelectedRange.location ||
       selectedRange.length != oldSelectedRange.length) {
     [self.inputDelegate selectionWillChange:self];
-    [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange]
+    [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange withText:self.text]
```

Comment:
Aren't we doing the newly added checks alreadyt on lines 206-209 above?

Reply:
Moved it out to a method